Add the ability to choose a random intro sequence each startup

V1ntagezTV opened the pull request:
Simple "Random" intro

- Closes #6284.

V1ntagezTV committed:
Add random intro!

V1ntagezTV committed:
remove needless blank

elleyer commented:
Any reason for that? Also, you should delete not English comments.

bdach commented:
@elleyer The submitter aims to resolve #6284 with this.

bdach commented on osu.Game/Screens/Loader.cs:
```diff
@@ -58,9 +59,16 @@ protected virtual OsuScreen CreateLoadableScreen()
 
         private IntroScreen getIntroSequence()
-        {
+        {//вот именно что не показывает ни всплывающих подсказок нихера
```
Please keep all source, including comments, in English. Additionally your whitespace needs fixing (as reported by CI).

bdach commented on osu.Game/Screens/Loader.cs:
```diff
@@ -58,9 +59,16 @@ protected virtual OsuScreen CreateLoadableScreen()
         private IntroScreen getIntroSequence()
-        {
+        {//вот именно что не показывает ни всплывающих подсказок нихера
+            Random random = new Random();
```
* Use `var` here.
* This allocation can be moved into the `switch` case.

V1ntagezTV committed:
remove comment

V1ntagezTV committed:
moved into the switch case

frenzibyte reviewed (changes requested):
This can be fairly simplified by just picking a random intro sequence enum value.

```csharp
switch (introSequence)
{
    case IntroSequence.Random:
        var introSequences = Enum.GetValues(typeof(IntroSequence)).Except(new[] { IntroSequence.Random }).ToList();
        return getIntroSequence(introSequences[RNG.Next(introSequences.Count)]);
}
```

Notes to be took from the above snippet:
 - `RNG` is used instead of `Random` which is a simplified way of seedless randomizing.

peppy committed:
Merge remote-tracking branch 'upstream/master' into add-random-intro

peppy committed:
Simplify and future-proof random retrieval method

Will support future added intros without further code changes. Also uses RNG instead of `new Random`.

smileyhead commented:
Nice, thanks! :-)
